core/test/mpi/base/polymorphic_object.cpp

```diff
@@ -0,0 +1,360 @@
+        const gko::Executor*, const gko::PolymorphicObject*,
+        const gko::PolymorphicObject*) const override
+    {
+        copy_started++;
+    }
+
+    void on_polymorphic_object_copy_completed(
+        const gko::Executor*, const gko::PolymorphicObject*,
+        const gko::PolymorphicObject*) const override
+    {
+        copy_completed++;
+    }
+
+    void on_polymorphic_object_move_started(
+        const gko::Executor*, const gko::PolymorphicObject*,
+        const gko::PolymorphicObject*) const override
+    {
+        move_started++;
+    }
+
+    void on_polymorphic_object_move_completed(
+        const gko::Executor*, const gko::PolymorphicObject*,
+        const gko::PolymorphicObject*) const override
+    {
+        move_completed++;
+    }
+
+    void on_polymorphic_object_deleted(
+        const gko::Executor*, const gko::PolymorphicObject*) const override
+    {
+        deleted++;
+    }
+
+    mutable int create_started = 0;
+    mutable int create_completed = 0;
+    mutable int copy_started = 0;
+    mutable int copy_completed = 0;
+    mutable int move_started = 0;
+    mutable int move_completed = 0;
+    mutable int deleted = 0;
+};
+
+
+struct DummyDistributedObject
+    : gko::experimental::EnableDistributedPolymorphicObject<
+          DummyDistributedObject>,
+      gko::EnableCreateMethod<DummyDistributedObject>,
+      gko::EnablePolymorphicAssignment<DummyDistributedObject>,
+      gko::experimental::distributed::DistributedBase {
+    explicit DummyDistributedObject(std::shared_ptr<const gko::Executor> exec,
+                                    gko::experimental::mpi::communicator comm,
+                                    int v = {})
+        : gko::experimental::EnableDistributedPolymorphicObject<
+              DummyDistributedObject>(std::move(exec)),
+          gko::experimental::distributed::DistributedBase(std::move(comm)),
+          x{v}
+    {}
+
+    DummyDistributedObject(const DummyDistributedObject& other)
+        : DummyDistributedObject(other.get_executor(), other.get_communicator())
+    {
+        *this = other;
+    }
+
+    DummyDistributedObject(DummyDistributedObject&& other)
+        : DummyDistributedObject(other.get_executor(), other.get_communicator())
+    {
+        *this = std::move(other);
+    }
+
+    DummyDistributedObject& operator=(const DummyDistributedObject& other)
+    {
+        if (this != &other) {
+            x = other.x;
+        }
+        return *this;
+    }
+
+    DummyDistributedObject& operator=(DummyDistributedObject&& other) noexcept
+    {
+        if (this != &other) {
+            x = std::exchange(other.x, 0);
+        }
+        return *this;
+    }
+
+    int x;
+};
+
+
+class EnableDistributedPolymorphicObject : public testing::Test {
+protected:
+    std::shared_ptr<gko::ReferenceExecutor> ref{
+        gko::ReferenceExecutor::create()};
+    std::shared_ptr<gko::OmpExecutor> omp{gko::OmpExecutor::create()};
```

Comment:
Currently, I don't think you can rely on `OmpExecutor` being usable (as `GINKGO_BUILD_OMP` is not dependent on `GINKGO_BUILD_MPI`). Since this is a `core` test, you can only rely on `ReferenceExecutor` being present (and the presence of MPI as the test is otherwise not built).
Weirdly enough, we do the same thing in the [`core/test/base/polymorphic_object.cpp`](https://github.com/ginkgo-project/ginkgo/blob/develop/core/test/base/polymorphic_object.cpp) file, We also need the `OmpExecutor` in the `core/test/base/{array,utils,allocator,executor}.cpp` tests, so this is not unique to `polymorphic_object`.
As we have the same issue in our code already, I am fine with this being ignored, but we should create an issue about that at least to fix it later on.

Reply:
also, dpcpp module can not find omp by default

Reply:
Since we already do this, as Thomas mentioned, I would also keep this here, but add a todo to fix that later.